Apps/Src/Voltage.c

```diff
@@ -90,9 +91,21 @@ void Voltage_UpdateMeasurements(void){
     LTC6811_rdcv_safe(0, NUM_MINIONS, Minions); // Set to read back all cell voltage registers
     //copies values from cells.c_codes to private array
 
+     // which minion we are currently sampling from
+    uint8_t moduleIdx = 0;
+    // the number of modules remaining to sample in the current minion
+    uint8_t minionModulesRemaining = VOLT_TAP_DIST[moduleIdx];
+
     // package raw voltage values into single array
     for(uint8_t i = 0; i < NUM_BATTERY_MODULES; i++){
-        rawVoltages[i] = Minions[i / MAX_VOLT_SENSORS_PER_MINION_BOARD].cells.c_codes[i % MAX_VOLT_SENSORS_PER_MINION_BOARD];
+
+        // if we are out of modules in this minion to sample, move to the next minion
+        if(minionModulesRemaining <= 0) {
+            moduleIdx ++;
+            minionModulesRemaining = VOLT_TAP_DIST[moduleIdx];
+        }
+        rawVoltages[i] = Minions[moduleIdx].cells.c_codes[VOLT_TAP_DIST[moduleIdx] - minionModulesRemaining];
```

Comment:
This looks like it works, but honestly i cant see why we cant just do 3 `memcpy` into rawVoltages from the Minions.cells.c_codes array. It might be cleaner too.

Reply:
can't memcpy cuz datatype mismatch. inner for loop that essentially performs a memcpy works tho, would def be cleaner

Reply:
yeah u right